sdk/identity/azure-identity/src/default_azure_credential.cpp

```diff
@@ -38,17 +43,46 @@ DefaultAzureCredential::DefaultAzureCredential(
             "is the better fit for the application.");
 
   // Creating credentials in order to ensure the order of log messages.
-  auto const envCred = std::make_shared<EnvironmentCredential>(options);
-  auto const wiCred = std::make_shared<WorkloadIdentityCredential>(options);
-  auto const azCliCred = std::make_shared<AzureCliCredential>(options);
-  auto const managedIdentityCred = std::make_shared<ManagedIdentityCredential>(options);
+  ChainedTokenCredential::Sources miSources;
+  {
+    miSources.emplace_back(std::make_shared<EnvironmentCredential>(options));
+    miSources.emplace_back(std::make_shared<WorkloadIdentityCredential>(options));
+
+    constexpr auto envVarName = "AZURE_TOKEN_CREDENTIALS";
+    const auto envVarValue = Environment::GetVariable(envVarName);
+
+    const auto isProd = StringExtensions::LocaleInvariantCaseInsensitiveEqual(envVarValue, "prod");
+    const auto logMsg = GetCredentialName() + ": '" + envVarName + "' environment variable is "
+        + (envVarValue.empty() ? "not set" : ("set to '" + envVarValue + "'"))
+        + ", therefore AzureCliCredential will " + (isProd ? "NOT " : "")
+        + "be included in the credential chain.";
+
+    if (isProd)
+    {
+      IdentityLog::Write(IdentityLog::Level::Verbose, logMsg);
+    }
+    else if (
+        envVarValue.empty()
```

Comment:
I'll be honest, the default being "enable the cli credential" doesn't seem like the best choice, but I recognize that's what the change requires and it does avoid a breaking change.

Reply:
Yes, that is exactly the intent - if you read #6590, that is the whole logic behind it, and all the SDKs now implemented this approach. Backwards compatible. It is still the easiest to grab credential with has the highest chances to get you authenticated as you write an Azure SDK app, and it is the first day of you doing this, so you don't want to figure out identity just yet. But now you, or your team have an option to exclude dev credentials even if you end up deploying DAC to prod, instead of using ManagedIdentityCredential, therefore reducing the risk.